### DISCUSSION TITLE
Active failover: make Database handler always run queries against a follower

### DISCUSSION THREAD

**Discussion by matcho** [arangodb/arangojs#790] [General]
Hi,

We currently have a setup with a realtime replication between 2 machines, where some read-only but load-heavy queries are explicitly run against the follower only.

Now we're considering switching to an active-failover setup with the same 2 machines, to achieve some kind of high availability.

Documentation says that ArangoJS connection pool can and will automatically run queries against the current leader, which is fine in most situations.

But in our case we would need the opposite: automatically run queries against the current follower.

Is there some way to achieve this ?

Thanks a lot,
Greetings,

Mathias

**Comment by chethan27** [comment]
can i work on this feature ?

**Comment by ghost** [comment]
You can use `allowDirtyRead` to round-robin across all servers _including_ followers. There's currently no flag for exclusively using dirty reads.
